### PR TITLE
Add methods for droplet and package management

### PR DESCRIPTION
**Pull request by IvanBorislavovDimitrov**
Get current droplet for application
Get package by id
Get packages by application id

Issue: LMCROSSITXSADEPLOY-2040

### CODE REVIEW

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudControllerClient.java [hunk 1]
```diff
@@ -901,4 +903,9 @@ void unbindServiceInstance(String applicationName, String serviceInstanceName,
 
     void updateServiceInstanceMetadata(UUID guid, Metadata metadata);
 
+    DropletInfo getCurrentDropletForApplication(UUID applicationGuid);
+
+    CloudPackage getPackage(UUID guid);
```
Comment (radoslav-d) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudControllerClient.java:908]: `CloudPackage getPackage(UUID packageGuid);`?

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudControllerClientImpl.java [hunk 2]
```diff
@@ -789,6 +791,21 @@ public void bindDropletToApp(UUID dropletGuid, UUID applicationGuid) {
         return handleExceptions(() -> delegate.getBuildsForPackage(packageGuid));
     }
 
+    @Override
+    public DropletInfo getCurrentDropletForApplication(UUID applicationGuid) {
+        return handleExceptions(() -> delegate.getCurrentDropletForApplication(applicationGuid));
+    }
+
+    @Override
+    public CloudPackage getPackage(UUID guid) {
```
Comment (radoslav-d) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/CloudControllerClientImpl.java:800]: `packageGuid`?

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/domain/DropletInfo.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+public interface DropletInfo {
+
+    @Nullable
+    @Value.Parameter
+    UUID getGuid();
+
+    @Nullable
+    @Value.Parameter
+    UUID getPackageGuid();
+}
```
Comment (radoslav-d) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/domain/DropletInfo.java:22]: nice!

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClient.java [hunk 4]
```diff
@@ -297,4 +299,9 @@
 
     void updateServiceInstanceMetadata(UUID guid, Metadata metadata);
 
+    DropletInfo getCurrentDropletForApplication(UUID applicationGuid);
+
+    CloudPackage getPackage(UUID guid);
```
Comment (radoslav-d) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClient.java:304]: again I think `packageGuid` is more suitable

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java [hunk 5]
```diff
@@ -1397,6 +1403,37 @@ public void bindDropletToApp(UUID dropletGuid, UUID applicationGuid) {
...
+    public DropletInfo getCurrentDropletForApplication(UUID applicationGuid) {
+        GetApplicationCurrentDropletResponse getApplicationCurrentDropletResponse = delegate.applicationsV3()
+                                                                                            .getCurrentDroplet(GetApplicationCurrentDropletRequest.builder()
+                                                                                                                                                  .applicationId(applicationGuid.toString())
+                                                                                                                                                  .build())
+                                                                                            .block();
+        String packageUrl = getApplicationCurrentDropletResponse.getLinks()
+                                                                .get("package")
+                                                                .getHref();
+        String packageGuid = packageUrl.substring(packageUrl.lastIndexOf("/") + 1);
```
Comment (radoslav-d) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java:1416]: Is this mechanism fail-safe?

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java [hunk 6]
```diff
@@ -1397,6 +1403,37 @@ public void bindDropletToApp(UUID dropletGuid, UUID applicationGuid) {
...
 
+    @Override
+    public DropletInfo getCurrentDropletForApplication(UUID applicationGuid) {
+        GetApplicationCurrentDropletResponse getApplicationCurrentDropletResponse = delegate.applicationsV3()
+                                                                                            .getCurrentDroplet(GetApplicationCurrentDropletRequest.builder()
+                                                                                                                                                  .applicationId(applicationGuid.toString())
+                                                                                                                                                  .build())
+                                                                                            .block();
+        String packageUrl = getApplicationCurrentDropletResponse.getLinks()
+                                                                .get("package")
```
Comment (radoslav-d) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java:1414]: Is there an enum or a dedicated constant for this `"package"` string?

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java (hunk 5 above)
Comment (IvanBorislavovDimitrov) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java]: A droplet is always bound to a package. A package could have more than one droplet so I assume it is safe. The response should always have link to the package: https://v3-apidocs.cloudfoundry.org/version/3.86.0/index.html#get-current-droplet

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java (hunk 6 above)
Comment (IvanBorislavovDimitrov) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java]: Moved in Contants

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java [hunk 7]
```diff
@@ -1397,6 +1404,37 @@ public void bindDropletToApp(UUID dropletGuid, UUID applicationGuid) {
...
     }
 
+    @Override
+    public DropletInfo getCurrentDropletForApplication(UUID applicationGuid) {
+        GetApplicationCurrentDropletResponse getApplicationCurrentDropletResponse = delegate.applicationsV3()
+                                                                                            .getCurrentDroplet(GetApplicationCurrentDropletRequest.builder()
+                                                                                                                                                  .applicationId(applicationGuid.toString())
+                                                                                                                                                  .build())
+                                                                                            .block();
+        String packageUrl = getApplicationCurrentDropletResponse.getLinks()
```
Comment (theghost5800) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java:1414]: Is it possible to get package guid with another method without custom parse of href link?

Comment (IvanBorislavovDimitrov) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java:1414]: I think it is possible to get all created builds for an application, then filter by droplet id, get the build and get the package id through the build. In my opinion, this is at least one additional call to the controller and requires more processing.

Comment (theghost5800) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java:1414]: If href link ends with slash `/` then parse logic will fail. We have to think can we return null object and create new package if we couldn't parse existing one.

File: cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java [hunk 8]
```diff
@@ -1397,6 +1408,53 @@ public void bindDropletToApp(UUID dropletGuid, UUID applicationGuid) {
...
+                                                                                            .block();
+        return tryParseDropletInfo(getApplicationCurrentDropletResponse);
+    }
+
+    private DropletInfo tryParseDropletInfo(GetApplicationCurrentDropletResponse getApplicationCurrentDropletResponse) {
+        try {
+            return parseDropletInfo(getApplicationCurrentDropletResponse);
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+            throw new CloudOperationException(HttpStatus.NOT_FOUND);
```
Comment (radoslav-d) [inline cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java:1426]: Is this the most suitable status code?

Comment (IvanBorislavovDimitrov) [reply cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java:1426]: roughly, but it is the safest way

**Pull request merged**